yt_dlp/extractor/bibeltv.py

```diff
@@ -1,27 +1,204 @@
-        'md5': '252f908192d611de038b8504b08bf97f',
+        'url': 'https://www.bibeltv.de/mediathek/videos/344436-alte-wege',
+        'md5': 'ec1c07efe54353780512e8a4103b612e',
         'info_dict': {
-            'id': 'ref:329703',
+            'id': '344436',
             'ext': 'mp4',
-            'title': 'Sprachkurs in Malaiisch',
-            'description': 'md5:3e9f197d29ee164714e67351cf737dfe',
-            'timestamp': 1608316701,
-            'uploader_id': '5840105145001',
-            'upload_date': '20201218',
-        }
+            'title': 'Alte Wege',
+            'description': 'md5:2f4eb7294c9797a47b8fd13cccca22e9',
+            'timestamp': 1677877071,
+            'upload_date': '20230303',
+            'view_count': int,
+            'episode': 'Episode 1',
+            'thumbnail': r're:https://bibeltv\.imgix\.net/[\w-]+\.jpg',
+            'like_count': int,
+            'episode_number': 1,
+            'duration': 150.0,
+        },
+        'params': {
+            'format': '6',
+        },
     }, {
         'url': 'https://www.bibeltv.de/mediathek/videos/crn/326374',
         'only_matching': True,
     }]
-    BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/5840105145001/default_default/index.html?videoId=ref:%s'
+    API_URL = 'https://www.bibeltv.de/mediathek/api'
+    AUTH_TOKEN = 'j88bRXY8DsEqJ9xmTdWhrByVi5Hm'
+
+    def _extract_formats_and_subtitles(self, data, crn_id, *, is_live=False):
+        formats = []
+        subtitles = {}
+        for media in data:
+            media_url = media['src']
+            media_ext = determine_ext(media_url)
+            if media_ext == 'm3u8':
+                m3u8_formats, m3u8_subs = self._extract_m3u8_formats_and_subtitles(
+                    media_url, crn_id, live=is_live)
+                formats.extend(m3u8_formats)
+                subtitles.update(m3u8_subs)
+            elif media_ext == 'mpd':
+                mpd_formats, mpd_subs = self._extract_mpd_formats_and_subtitles(media_url, crn_id)
+                formats.extend(mpd_formats)
+                subtitles.update(mpd_subs)
+            elif media_ext == 'mp4':
+                formats.append({'url': media['src']})
+            else:
+                self.report_warning(f'Unknown format {media_ext!r}')
+
+        return formats, subtitles
+
+    @staticmethod
+    def _extract_url_info(data):
+        crn_id = data['crn']
+
+        return {
+            '_type': 'url',
+            'id': crn_id,
+            'url': f'https://www.bibeltv.de/mediathek/videos/{data.get("slug")}',
+            'title': data.get('title'),
+            'description': data.get('description'),
+            'duration': try_get(data, lambda x: x['duration'] / 1000),
+            'timestamp': parse_iso8601(data.get('schedulingStart')),
+            'thumbnails': [{'url': url} for url in {image['url'] for image in data.get('images', ())}],
```

Comment:
This might be cleaner? Are there direct duplicates?
```suggestion
            'thumbnails': [{'url': url} for url in set(traverse_obj(data, ('images', ..., 'url', {url_or_none})))],
```

Reply:
wouldn't this work?
```suggestion
            'thumbnails': orderedSet(traverse_obj(data, ('images', ..., {'url': ('url', {url_or_none}}))),
```

Reply:
> [...] Are there direct duplicates?
Yes

Reply:
> wouldn't this work?
> ```suggestion
>             'thumbnails': orderedSet(traverse_obj(data, ('images', ..., {'url': ('url', {url_or_none}}))),
> ```

Right, bc empty dict is discarded, mb. There is a syntactical error in there but otherwise preferred way then